modules/ublog/src/main/UblogApi.scala

```diff
@@ -201,3 +202,61 @@ final class UblogApi(
     !u.isBot && {
       (u.count.game > 0 && u.createdSinceDays(2)) || u.hasTitle || u.isVerified || u.isPatron
     }
+
+  // So far this only hits a prod index if $select contains `topics`, or if byDate is false
+  // i.e. byDate can only be true if $select contains `topics`
+  private[ublog] def aggregateVisiblePosts(
+      select: Bdoc,
+      offset: Int,
+      length: Int,
+      ranking: UblogRank.Sorting = UblogRank.Sorting.ByRank
+  ) =
+    colls.post
+      .aggregateList(length, _.sec): framework =>
+        import framework.*
+        Match(select ++ $doc("live" -> true)) -> (ranking.sortingQuery(colls.post, framework) ++
+          List(Limit(100))
+          ++ removeUnlistedOrClosedAndProjectForPreview(colls.post, framework) ++ List(
+            Skip(offset),
+            Limit(length)
+          ))
+      .map: docs =>
+        for
+          doc  <- docs
+          post <- doc.asOpt[UblogPost.PreviewPost]
+        yield post
+
+  private[ublog] def removeUnlistedOrClosedAndProjectForPreview(
```

Comment:
I initially tried the function's name not to be too java-y but it's hard to find a shorter way to keep all meaning.

Reply:
I'd rather have a java-y function name, than a function silently doing more than it seems